Python 3 compatibility issue with `self_generated_targets`

### Description

I get `AssertionError` when trying to return `self_generated_targets` in a model.

### Environment information

```
OS: macOS 10.13.4

$ pip freeze | grep tensor
mesh-tensorflow==0.0.4
tensor2tensor==1.11.0
tensorboard==1.12.0
tensorflow==1.12.0
tensorflow-metadata==0.9.0
tensorflow-probability==0.5.0

$ python -V
Python 3.6.4
```

### For bugs: error logs

```
  File "/Users/ywkim/tensor2tensor/tensor2tensor/utils/t2t_model.py", line 1266, in wrapping_model_fn                                                                 
    use_tpu=use_tpu)
  File "/Users/ywkim/tensor2tensor/tensor2tensor/utils/t2t_model.py", line 1340, in estimator_model_fn                                                                
    "Found {}".format(logits.keys())
AssertionError: Expect only key 'logits' when there is 'self_generated_targets'. Found dict_keys(['logits'])
```

--- a/tensor2tensor/utils/t2t_model.py
+++ b/tensor2tensor/utils/t2t_model.py
@@ -1337,7 +1337,7 @@ class T2TModel(base.Layer):
       # by logits["self_generated_targets"].
       tf.logging.info("Replacing targets with model-provided targets.")
       features["targets"] = labels = logits.pop("self_generated_targets")
-      assert logits.keys() == ["logits"], (
+      assert list(logits.keys()) == ["logits"], (
           # See "Returns" in the "top" method docstring for the expected
           # "logits" format when targets are generated at training time.
           "Expect only key 'logits' when there is 'self_generated_targets'. "

